packages/jsapi-bootstrap/src/useWidget.test.tsx

```diff
@@ -78,30 +66,119 @@ describe('useWidget', () => {
+  it('should close the widget and exported objects when cancelled', async () => {
+    const descriptor = { type: WIDGET_TYPE, name: 'name' };
+    const widget: dh.Widget = TestUtils.createMockProxy<dh.Widget>({
+      close: jest.fn(),
+      exportedObjects: [
+        TestUtils.createMockProxy<dh.WidgetExportedObject>({
+          close: jest.fn(),
+        }),
+        TestUtils.createMockProxy<dh.WidgetExportedObject>({
+          close: jest.fn(),
+        }),
+      ],
+    });
+    const fetch = jest.fn(async () => widget);
+    const objectFetch = { fetch, error: null };
+    const subscribe = jest.fn((_, onUpdate) => {
+      onUpdate(objectFetch);
+      return jest.fn();
+    });
+    const objectManager = { subscribe };
+    const wrapper = ({ children }) => (
+      <ObjectFetchManagerContext.Provider value={objectManager}>
+        {children}
+      </ObjectFetchManagerContext.Provider>
+    );
+    const { result, unmount } = renderHook(() => useWidget(descriptor), {
+      wrapper,
+    });
+    expect(widget.close).not.toHaveBeenCalled();
+    expect(widget.exportedObjects[0].close).not.toHaveBeenCalled();
+    expect(widget.exportedObjects[1].close).not.toHaveBeenCalled();
+
+    expect(result.current).toEqual({ widget: null, error: null });
+
+    // Unmount before flushing the promise
+    unmount();
+    await act(TestUtils.flushPromises);
+    expect(widget.close).toHaveBeenCalledTimes(1);
+    expect(widget.exportedObjects[0].close).toHaveBeenCalledTimes(1);
+    expect(widget.exportedObjects[1].close).toHaveBeenCalledTimes(1);
+  });
+
+  it('should not close the widget if it is returned before unmount', async () => {
+    const descriptor = { type: WIDGET_TYPE, name: 'name' };
+    const widget: dh.Widget = TestUtils.createMockProxy<dh.Widget>({
+      close: jest.fn(),
+      exportedObjects: [
+        TestUtils.createMockProxy<dh.WidgetExportedObject>({
+          close: jest.fn(),
+        }),
+        TestUtils.createMockProxy<dh.WidgetExportedObject>({
+          close: jest.fn(),
+        }),
+      ],
+    });
+    const fetch = jest.fn(async () => widget);
+    const objectFetch = { fetch, error: null };
+    const subscribe = jest.fn((_, onUpdate) => {
+      onUpdate(objectFetch);
+      return jest.fn();
+    });
+    const objectManager = { subscribe };
+    const wrapper = ({ children }) => (
+      <ObjectFetchManagerContext.Provider value={objectManager}>
+        {children}
+      </ObjectFetchManagerContext.Provider>
+    );
+    const { result, unmount } = renderHook(() => useWidget(descriptor), {
+      wrapper,
+    });
+
+    expect(result.current).toEqual({ widget: null, error: null });
+    await act(TestUtils.flushPromises);
+    unmount();
+    expect(widget.close).not.toHaveBeenCalled();
+    expect(widget.exportedObjects[0].close).not.toHaveBeenCalled();
+    expect(widget.exportedObjects[1].close).not.toHaveBeenCalled();
+  });
+
+  it('should handle a Table being fetched', async () => {
```

Comment:
Do we want a test for closing a table too? Idk if it's necessary since it would just be skipping the `exportedObjects` close block if it were a table

Reply:
May as well